src/client/datascience/jupyter/jupyterServer.ts

```diff
@@ -144,7 +144,7 @@ export class JupyterServerBase implements INotebookServer {
         this.asyncRegistry.push(this);
     }
 
-    public async connect(launchInfo: INotebookServerLaunchInfo, cancelToken?: CancellationToken): Promise<void> {
+    public async connect(launchInfo: INotebookServerLaunchInfo, cancelToken?: CancellationToken, connectTimeout?: number): Promise<void> {

```

Comment:
Why not add a default value for `connectTimeout`
```suggestion
    public async connect(launchInfo: INotebookServerLaunchInfo, cancelToken?: CancellationToken, connectTimeout: number = Number.MAX_SAFE_INTEGER): Promise<void> {
```

With this done, then you don't need to check if this arg is empty (further below).